Source/Particles/Gather/FieldGather.H

```diff
@@ -71,10 +71,10 @@ void doGatherShapeN (const amrex::ParticleReal xp,
     amrex::ignore_unused(n_rz_azimuthal_modes);
 #endif
 
-    const amrex::Real dxi = 1.0/dx[0];
-    const amrex::Real dzi = 1.0/dx[2];
+    const amrex::Real dxi = 1.0_rt/dx[0];
+    const amrex::Real dzi = 1.0_rt/dx[2];
 #if (AMREX_SPACEDIM == 3)
-    const amrex::Real dyi = 1.0/dx[1];
+    const amrex::Real dyi = 1.0_rt/dx[1];
```

Comment:
In the previous version, with `amrex::Real` = `float` , the right hand side was calculated in `double` precision and then converted to `float` . I might be wrong, but I guess that this was not done on purpose...